src/System.Private.CoreLib/shared/System/Threading/Tasks/Task.cs

```diff
@@ -6110,6 +6070,9 @@ internal static Task<TResult> CreateUnwrapPromise<TResult>(Task outerTask, bool
             return new UnwrapPromise<TResult>(outerTask, lookForOce);
         }
 
+#if CORERT
+        [DependencyReductionRoot]
+#endif
```

Comment:
@jkotas, rather than having ifdefs around DependencyReductionRoot, could we define one in coreclr?  It could even be attributed as Conditional so that it's compiled away in release builds.

Reply:
There are not that many and most of them are in places that are different between CoreCLR and ProjectN for other reasons.

If somebody want to do it, I would not mind, but I am not sure whether it is worth it. I would love for the `DependencyReductionRoot` attribute to go away eventually.